ARCs/arc-0056.md

```diff
@@ -272,9 +272,12 @@ type StructName = string;
 /** Raw byteslice without the length prefixed that is specified in ARC-4 */
 type AVMBytes = "bytes";
 
-/** Mapping of named structs to the ABI type of their fields */
-interface StructFields {
-  [fieldName: string]: ABIType | StructFields;
+/** Information about a single field in a struct */
+interface StructField = {
+  /** The name of the struct field */
+  name: string;
+  /** The type of the struct field's value */
+  type: ABIType | StructName;
```

Comment:
Should this be `ABIType | StructField[]` otherwise you have to create a unique name for all sub-values and makes the recursive logic harder?

Reply:
Maybe `ABIType | StructName | StructField[]` as in some cases structs may be reused? And for languages without anonymous structured types (e.g. python) there should be a unique name already

Reply:
Ah yeah, fair